common/src/main/java/com/wynntils/services/mapdata/MapFeatureRenderer.java

```diff
@@ -185,6 +199,59 @@ public static void renderMapLocation(
         poseStack.popPose();
     }
 
+    private static void renderMapArea(
+            PoseStack poseStack,
+            MultiBufferSource bufferSource,
+            MapArea area,
+            ResolvedMapAttributes attributes,
+            Vector2f mapCenter,
+            Vector2f screenCenter,
+            Vector2f rotationVector,
+            float zoomLevel,
+            float zoomRenderScale,
+            float featureRenderScale,
+            boolean hovered,
+            boolean fullscreenMap) {
+        List<Vector2f> worldVertices = area.getBoundingPolygon().vertices();
+
+        // Transform area vertices to screen coordinates
+        List<Vector2f> screenVertices = worldVertices.stream()
+                .map(vertex -> getRenderLocation(mapCenter, screenCenter, rotationVector, zoomRenderScale, vertex))
+                .toList();
+
+        BufferedRenderUtils.drawPolygon(
+                poseStack,
+                bufferSource,
+                attributes.fillColor(),
+                attributes.borderColor(),
+                attributes.borderWidth(),
+                screenVertices,
+                0);
+
+        BoundingPolygon boundingPolygon = BoundingPolygon.fromVertices(screenVertices);
+        Vector2f centroid = boundingPolygon.centroid();
+
+        int labelWidth = (int)
```

Comment:
I get the feeling this code should be possible to share with drawing labels for locations. Or is it slightly different enough that it is not worth the effort to try and unite them?

Reply:
I only like to "dry out code" if it is repeated 3 times. This is only the second case :).

Reply:
Kind of depends on how long the repeated piece of code is. But okay...